### PR TITLE
JBPM-4767: Process/Task dashboard details screen (in log mode) switch…

### PR DESCRIPTION
**Pull request by nmirasch**
… to default after refresh

### CODE REVIEW

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiPresenter.java [hunk 1]
```diff
@@ -106,6 +106,23 @@
 
     private String processId = "";
 
+    private boolean forLog =false;
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiPresenter.java]: please check code formatting

**Comment by cristianonicolai** [comment]
@nmirasch there is two things I found here, see image below:

selection_060

1 - The UI has changed as part of the PatternFly migration and logs and details are in different tabs now. You can notice that we're missing the details tab in here, please have a look on that.

2 - When selecting a task, the UI where the list is gets resized really weird.

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetails/TaskDetailsPresenter.java [hunk 2]
```diff
@@ -206,6 +185,15 @@ public boolean error( Message message,
...
 
+    public void disableTaskDetailEdition(){
+        view.getTaskDescriptionTextArea().setEnabled( false );
+        view.getDueDate().setEnabled( false );
+        view.getUserText().setEnabled( false );
+        view.getTaskStatusText().setEnabled( false );
+        view.getDueDateTime().setEnabled( false );
+        view.getTaskPriorityListBox().setEnabled( false );
+        view.getUpdateTaskButton().setVisible( false );
+    }
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetails/TaskDetailsPresenter.java]: please add a break line to separate both methods.

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiPresenter.java [hunk 3]
```diff
@@ -106,6 +106,10 @@
 
     private String processId = "";
 
+    private boolean forLog = false;
+
+    private boolean forAdmin = false ;
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiPresenter.java]: false;

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/test/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiViewImplTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @InjectMocks
+    private TaskDetailsMultiViewImpl taskDetailsMultiView;
+
+    @Mock
+    private TaskDetailsMultiPresenter presenterMock;
+
+    @Mock
+    NavTabs navTabs;
+
+   @Before
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/test/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiViewImplTest.java]: Please remove

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/test/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiPresenterTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    @Spy
+    Event<ChangeTitleWidgetEvent> changeTitleWidgetEvent = new EventSourceMock<ChangeTitleWidgetEvent>();
+
+    @Spy
+    Event<TaskSelectionEvent> taskSelectionEvent = new EventSourceMock<TaskSelectionEvent>();
+
+    @InjectMocks
+    private TaskDetailsMultiPresenter presenter;
+
+   @Before
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/test/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiPresenterTest.java]: please add another space to align `@Before` with other annotations.

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/test/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiViewImplTest.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Test
+    public void displayOnlyLogTabTest() {
+        taskDetailsMultiView.init(presenterMock);
+
+        taskDetailsMultiView.displayOnlyLogTab();
+        verify(presenterMock).disableTaskDetailsEdition();
+    }
+
+    public void initTabsTest(){
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/test/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiViewImplTest.java]: missing `@Test`

**Comment by cristianonicolai** [comment]
@nmirasch now that the details tab is showing, I notice that the priority select isn't set as disabled. 

selection_061

**Comment by cristianonicolai** [comment]
@nmirasch I also notice a few error logs in the console while opening/closing the details screen, not sure what those are and if you saw similar problem.

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiViewImpl.java [hunk 7]
```diff
@@ -225,9 +234,15 @@ public void displayOnlyLogTab() {
         for ( Widget active : tabContent ) {
             active.setVisible( false );
         }
+        taskDetailsPane.setVisible(true);
+        taskDetailsTab.setVisible(true);
+        presenter.disableTaskDetailsEdition();
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskdetailsmulti/TaskDetailsMultiViewImpl.java]: here is looping back to the presenter, instead I think you could this in the presenter itself right after calling view.displayOnlyLogTab ?

**Comment by cristianonicolai** [comment]
+1 thanks @nmirasch!

**Comment by psiroky** [comment]
@nmirasch could you please rebase this on top of the current 6.4.x and fix the merge conflicts (they were introduced after merging one of the other PRs)?

**Comment by nmirasch** [comment]
@psiroky done

**Comment by csadilek** [comment]
master: a6d803a7f83b6b082117916c9613461d044c3969

**Comment by csadilek** [comment]
6.4.x: 73bde63c879b7539fce567708df6a0064bbfd7ca

**Pull request closed without merging**